SDL/MacOSBarItems.mm

```diff
@@ -0,0 +1,326 @@
+@property (assign) NSMenu *openMenu;
+@property (assign) std::shared_ptr<I18NCategory> mainSettingsLocalization;
+@end
+
+void initBarItemsForApp() {
+    [[BarItemsManager sharedInstance] setupAppBarItems];
+}
+
+// im soooooo sorry for whoever had to read this impl
+@implementation BarItemsManager
++ (instancetype)sharedInstance {
+    static BarItemsManager *stub;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        stub = [BarItemsManager new];
+        stub.mainSettingsLocalization = GetI18NCategory("MainSettings");
+    });
+    
+    return stub;
+}
+
+-(void)setupAppBarItems {
+    NSMenuItem *openMenuItem = [[NSMenuItem alloc] init];
+    openMenuItem.submenu = [self makeOpenSubmenu];
+    
+    NSMenuItem *graphicsMenuItem = [[NSMenuItem alloc] init];
+    graphicsMenuItem.submenu = [self makeGraphicsMenu];
+    
+    NSMenuItem *audioMenuItem = [[NSMenuItem alloc] init];
+    audioMenuItem.submenu = [self makeAudioMenu];
+    
+    [NSApplication.sharedApplication.menu addItem:openMenuItem];
+    [NSApplication.sharedApplication.menu addItem:graphicsMenuItem];
+    [NSApplication.sharedApplication.menu addItem:audioMenuItem];
+}
+
+-(NSMenu *)makeOpenSubmenu {
+    NSMenu *menu = [[NSMenu alloc] initWithTitle:@"File"];
+    NSMenuItem *openWithSystemFolderBrowserItem = [[NSMenuItem alloc] initWithTitle:@"Open..." action:@selector(openSystemFileBrowser) keyEquivalent:@"o"];
+    openWithSystemFolderBrowserItem.keyEquivalentModifierMask = NSEventModifierFlagCommand;
+    openWithSystemFolderBrowserItem.enabled = YES;
+    openWithSystemFolderBrowserItem.target = self;
+    [menu addItem:openWithSystemFolderBrowserItem];
+    self.openMenu = menu;
+    
+    [self addOpenRecentlyItem];
+    return menu;
+}
+
+-(NSMenu *)makeGraphicsMenu {
+    NSMenu *parent = [[NSMenu alloc] initWithTitle:@(self.mainSettingsLocalization->T("Graphics"))];
+    NSMenu *backendsMenu = [[NSMenu alloc] init];
+    
+    auto graphicsLocalization = GetI18NCategory("Graphics");
+#define GRAPHICS_LOCALIZED(key) @(graphicsLocalization->T(key))
+    
+    NSMenuItem *gpuBackendItem = [[NSMenuItem alloc] initWithTitle:GRAPHICS_LOCALIZED("Backend") action:nil keyEquivalent:@""];
+    
+    std::vector<GPUBackend> allowed = [self allowedGPUBackends];
+    for (int i = 0; i < allowed.size(); i++) {
+        NSMenuItem *backendMenuItem = [[NSMenuItem alloc] initWithTitle:@(GPUBackendToString(allowed[i]).c_str()) action: @selector(setCurrentGPUBackend:) keyEquivalent: @""];
+        backendMenuItem.tag = i;
+        backendMenuItem.target = self;
+        backendMenuItem.state = [self controlStateForBool: g_Config.iGPUBackend == (int)allowed[i]];
+        [backendsMenu addItem:backendMenuItem];
+    }
+    
+    gpuBackendItem.submenu = backendsMenu;
+    [parent addItem:gpuBackendItem];
+    
+    [parent addItem:[NSMenuItem separatorItem]];
+    
+#define MENU_ITEM(variableName, localizedTitleName, SEL, ConfigurationValueName) \
```

Comment:
nit: It feels like this could just be a regular method that returns `NSMenuItem *`.  It also feels like it could automatically call `[parent addItem:] for you.  It's fine, just nitmumbling to myself.

-[Unknown]